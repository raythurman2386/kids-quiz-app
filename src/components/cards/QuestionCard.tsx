import React from 'react';
import { QuestionCardProps } from '../../types';
import { Wrapper, ButtonWrapper } from '../../styles/QuestionCard.styles';

const QuestionCard: React.FC<QuestionCardProps> = ({
  question,
  answers,
  callback,
  userAnswer,
  questionNum,
  totalQuestions,
}) => (
  <Wrapper data-testid='question-card-wrapper'>
    <p className='number'>
      Question: {questionNum} / {totalQuestions}
    </p>
    <p dangerouslySetInnerHTML={{ __html: question }} />
    <div>
      {answers.map(answer => (
        <ButtonWrapper
          correct={userAnswer?.correctAnswer === answer}
          userClicked={userAnswer?.answer === answer}
          key={answer}
        >
          <button disabled={!!userAnswer} value={answer} onClick={callback}>
            <span dangerouslySetInnerHTML={{ __html: answer }} />
          </button>
        </ButtonWrapper>
      ))}
    </div>
  </Wrapper>
);

export default QuestionCard;
