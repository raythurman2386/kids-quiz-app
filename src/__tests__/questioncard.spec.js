import '@testing-library/jest-dom/extend-expect';

import React from 'react';
import { render } from '@testing-library/react';

import QuestionCard from '../components/cards/QuestionCard';

describe('Question Card component tests', () => {
  test('Question Card component displays correctly.', () => {
    const card = {
      question: 'test question',
      answers: ['a', 'b', 'c', 'd'],
      callback: () => 'Hello',
      userAnswer: 'a',
      questionNum: 1,
      totalQuestions: 1,
    };
    const wrapper = render(
      <QuestionCard
        question={card.question}
        answers={card.answers}
        questionNum={card.questionNum}
        totalQuestions={card.totalQuestions}
        userAnswer={card.userAnswer}
        callback={card.callback}
      />
    );
    expect(wrapper.getByTestId('question-card-wrapper')).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
});
