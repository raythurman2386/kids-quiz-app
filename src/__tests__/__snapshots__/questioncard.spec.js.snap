// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question Card component tests Question Card component displays correctly. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnylx esHDrr"
        data-testid="question-card-wrapper"
      >
        <p
          class="number"
          data-testid="question-number"
        >
          Question: 
          1
           / 
          1
        </p>
        <p>
          test question
        </p>
        <div>
          <div
            class="sc-gtssRu gDxeRB"
          >
            <button
              disabled=""
              value="a"
            >
              <span>
                a
              </span>
            </button>
          </div>
          <div
            class="sc-gtssRu gDxeRB"
          >
            <button
              disabled=""
              value="b"
            >
              <span>
                b
              </span>
            </button>
          </div>
          <div
            class="sc-gtssRu gDxeRB"
          >
            <button
              disabled=""
              value="c"
            >
              <span>
                c
              </span>
            </button>
          </div>
          <div
            class="sc-gtssRu gDxeRB"
          >
            <button
              disabled=""
              value="d"
            >
              <span>
                d
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnylx esHDrr"
      data-testid="question-card-wrapper"
    >
      <p
        class="number"
        data-testid="question-number"
      >
        Question: 
        1
         / 
        1
      </p>
      <p>
        test question
      </p>
      <div>
        <div
          class="sc-gtssRu gDxeRB"
        >
          <button
            disabled=""
            value="a"
          >
            <span>
              a
            </span>
          </button>
        </div>
        <div
          class="sc-gtssRu gDxeRB"
        >
          <button
            disabled=""
            value="b"
          >
            <span>
              b
            </span>
          </button>
        </div>
        <div
          class="sc-gtssRu gDxeRB"
        >
          <button
            disabled=""
            value="c"
          >
            <span>
              c
            </span>
          </button>
        </div>
        <div
          class="sc-gtssRu gDxeRB"
        >
          <button
            disabled=""
            value="d"
          >
            <span>
              d
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
